dataset/src/main/scala/frameless/RecordEncoder.scala

```diff
@@ -168,6 +171,86 @@ class RecordEncoder[F, G <: HList, H <: HList]
       val newExpr = NewInstance(classTag.runtimeClass, newArgs, jvmRepr, propagateNull = true)
 
       val nullExpr = Literal.create(null, jvmRepr)
+
       If(IsNull(path), nullExpr, newExpr)
     }
 }
+
+final class RecordFieldEncoder[T](
```

Comment:
I think the `RecordFieldEncoder` makes sense to me now :+1: 
I wanted to avoid intoroducing new extra wrappers for derivation, but it is pretty elegant.